src/NHibernate/Linq/UpdateSyntax.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Remotion.Linq;
+
+namespace NHibernate.Linq
+{
+	public class UpdateSyntax<T>
+	{
+		private readonly IQueryable<T> _query;
+
+		internal UpdateSyntax(IQueryable<T> query)
+		{
+			_query = query;
+		}
+
+
+		/// <summary>
+		/// Specify the assignments and execute the update.
+		/// </summary>
+		/// <param name="assignments">The assignments.</param>
+		/// <param name="versioned">if set to <c>true</c> [versioned].</param>
+		/// <returns></returns>
+		public int Assign(Action<Assignments<T, T>> assignments, bool versioned = false)
```

Comment:
Shouldn't it be versioned by default for entity having versioning, if possible?  (I guess there are some hurdles indeed.)
The xml doc should maybe be written : `<c>true</c> for having the update increment the entities version, <c>false</c> otherwise.`

Instead of `Assign`, maybe `Values` would do.